pkg/kv/kvserver/store_snapshot.go

```diff
@@ -549,8 +549,15 @@ func (kvSS *kvBatchSnapshotStrategy) Receive(
 			}
 
 			timingTag.start("sst")
+			verifyCheckSum := snapshotChecksumVerification.Get(&s.ClusterSettings().SV)
 			// All batch operations are guaranteed to be point key or range key puts.
 			for batchReader.Next() {
+				// Verify value checksum to catch data corruption.
+				if verifyCheckSum {
+					if err = batchReader.Verify(); err != nil {
+						return noSnap, err
+					}
+				}
 				switch batchReader.KeyKind() {
```

Comment:
Are we paying twice for parsing the key above (in `Verify` func) and below? Is there a way to save here?